lua/cmp_nvim_lsp_signature_help/init.lua

```diff
@@ -138,13 +153,22 @@ source._docs = function(self, signature, parameter_index)
   return { kind = 'markdown', value = table.concat(documentation, '\n') }
 end
 
-source._signature_label = function(self, signature, parameter_index)
+source._signature_label = function(self, signature, parameter_index, opts)
   local label = signature.label
   if parameter_index then
-    local s, e = string.find(label, self:_parameter_label(signature, signature.parameters[parameter_index]), 1, true)
-    if s and e then
-      local active = string.sub(label, s, e)
-      label = string.gsub(label, vim.pesc(active), '***' .. active .. '***')
+    if opts.extended_docs then
+      local doc = signature.parameters[parameter_index].documentation
+      if doc.value then
+        label = doc.value
+      else
+        label = doc
+      end
+    else
+      local s, e = string.find(label, self:_parameter_label(signature, signature.parameters[parameter_index]), 1, true)
+      if s and e then
+        local active = string.sub(label, s, e)
+        label = string.gsub(label, vim.pesc(active), '***' .. active .. '***')
+      end
     end
```

Comment:
Thanks for your help!

I don't think you need the `extended_docs` option. Your implementation is correct. I just forgot to implement the parameter documentation.

Reply:
Ok, I've removed the config part and instead replaced the old logic. I hope that's what you meant?